protocol/contracts/nexus/Shard.sol

```diff
@@ -0,0 +1,489 @@
+     * @param minter Address of the minter
+     * @param shardId Shard ID
+     * @param allowed If minter is allowed to mint
+     */
+    function setMinterAllowedShardId(
+        address minter,
+        uint256 shardId,
+        bool allow
+    ) external onlyElevatedAccess {
+        _setMinterAllowedShardId(minter, shardId, allow);
+    }
+
+    function _setMinterAllowedShardId(
+        address minter,
+        uint256 shardId,
+        bool allow
+    ) internal {
+        if (minter == ZERO_ADDRESS) { revert ZeroAddress(); }
+        if (shardId >= _nextTokenId()) { revert CommonEventsAndErrors.InvalidParam(); }
+        if (allow) {
+            allowedShardIds[minter].add(shardId);
+        } else {
+            allowedShardIds[minter].remove(shardId);
+        }
+        emit MinterAllowedShardIdSet(minter, shardId, allow); 
+    }
+
+    /*
+     * @notice Set multiple shard IDs that minter can mint. This is an explicit setting.
+     * Shard ID must exist. To enable minter for a new shard, use functions setNewMinterShard and setNewMinterShards
+     * @param minter Address of the minter
+     * @param shardIds Shard IDs
+     * @param flags Booleans for if the partner can mint shards
+     */
+    function setMinterAllowedShardIds(
+        address minter,
+        uint256[] calldata shardIds,
+        bool[] calldata flags
+    ) external onlyElevatedAccess {
+        if (minter == ZERO_ADDRESS) { revert ZeroAddress(); }
+        uint256 _length = shardIds.length;
+        if (_length != flags.length) { revert InvalidParamLength(); }
+        bool allowed;
+        uint256 shardId;
+        for (uint i; i < _length;) {
+            allowed = flags[i];
+            shardId = shardIds[i];
+            if (shardId >= _nextTokenId()) { revert CannotMint(shardId); }
+            EnumerableSet.UintSet storage minterShardIds = allowedShardIds[minter];
+            if (allowed) {
+                minterShardIds.add(shardId);
+            } else {
+                minterShardIds.remove(shardId);
+            }
+            emit MinterAllowedShardIdSet(minter, shardId, allowed);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /*
+     * @notice Set the caps for shards minters can mint
+     * cap is 0 by default which means unlimited
+     * @param minter Address of the partner
+     * @param shardIds Shard IDs
+     * @param caps The maximum amount partner can mint for each shard
+     */
+    function setAllowedShardCaps(
+        address minter,
+        uint256[] calldata shardIds,
+        uint256[] calldata caps
+    ) external onlyElevatedAccess {
+        if (minter == ZERO_ADDRESS) { revert ZeroAddress(); }
+        uint256 _length = shardIds.length;
+        if (_length != caps.length) { revert InvalidParamLength(); }
+        uint256 shardId;
+        uint256 cap;
+        for (uint i; i < _length; ) {
+            shardId = shardIds[i];
+            cap = caps[i];
```

Comment:
Should this also check if the minter has access to mint this shard, where `allowedShardIds[minter].contains(shardId) == true`?